web/__tests__/pages/api/post/post.int.test.ts

```diff
@@ -181,112 +183,101 @@ describe("[API] Post - Integration Test", () => {
     });
 
     test("singleton covered-uncovered", async () => {
-      const coveredPost = createRandomPost();
-      coveredPost["covered"] = true;
-      let id = new mongoose.Types.ObjectId();
+      const post = createRandomPost();
+      const coveredId = new mongoose.Types.ObjectId();
       const createCoveredPost = await Post.create({
-        ...coveredPost,
-        _id: id,
-        attachments,
+        ...post,
+        _id: coveredId,
+        covered: true,
       });
       expect(createCoveredPost).not.toBeNull();
-      coveredPost["covered"] = false;
-      id = new mongoose.Types.ObjectId();
+      const uncoveredId = new mongoose.Types.ObjectId();
       const createUncoveredPost = await Post.create({
-        ...coveredPost,
-        _id: id,
-        attachments,
+        ...post,
+        _id: uncoveredId,
+        covered: false,
       });
       expect(createUncoveredPost).not.toBeNull();
-      let goodWithArray = [];
-      for (const goodWithKey in goodWithMap) {
-        if (createCoveredPost[goodWithKey] == "yes") {
-          goodWithArray.push(goodWithMap[goodWithKey]);
-        }
-      }
-      const postFilterSchema = {
-        type: [coveredPost.type],
-        breed: coveredPost.breed,
-        age: [coveredPost.age],
-        size: [coveredPost.size],
-        gender: [coveredPost.gender],
-        behavioral: coveredPost.behavioral,
-        goodWith: goodWithArray,
+      const postFilters = {
+        type: Object.values(FosterType),
+        breed: Object.values(Breed),
+        age: Object.values(Age),
+        size: Object.values(Size),
+        gender: Object.values(Gender),
+        behavioral: Object.values(Behavioral),
+        goodWith: [],
       };
       const filteredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
       });
       expect(filteredPosts).not.toBeNull();
       expect(filteredPosts.length).toBe(2);
       const coveredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
         covered: true,
       });
       expect(coveredPosts).not.toBeNull();
       expect(coveredPosts.length).toBe(1);
+      expect(coveredPosts[0]._id).toMatchObject(coveredId);
       const uncoveredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
         covered: false,
       });
       expect(uncoveredPosts).not.toBeNull();
       expect(uncoveredPosts.length).toBe(1);
+      expect(uncoveredPosts[0]._id).toMatchObject(uncoveredId);
     });
 
     test("random covered-uncovered", async () => {
-      let randomPost = createRandomPost();
-      const numPosts = Math.ceil(Math.random() * 10);
-      let numCovered = 0;
-      let numUncovered = 0;
+      const randPosts = randomPosts;
+      let allIds = [];
+      let covered = [];
+      let uncovered = [];
```

Comment:
You don't need to maintain an array for ids, covered, and uncovered posts - just have expect().toMatchObject() with the filtered arrays handle the checking